client/src/modules/stock/entry/entry.js

```diff
@@ -553,9 +554,50 @@ function StockEntryController(
 
     movement.lots = Stock.processLotsFromStore(vm.stockForm.store.data, vm.movement.entity.uuid);
 
-    // fix the unit price, if necessary
+    const exchangeRate = Exchange.getCurrentRate(vm.currencyId);
+
+    // If there are shipping and handling charges, distribute them to the individual lot articles
+    if (vm.originalPO[0].shipping_handling) {
+
+      // Compute the cost of the original order (in PO currency), not including shipping+handling
+      const costPO = vm.originalPO.reduce((accum, lot) => accum + lot.unit_price * lot.quantity, 0.0);
+
+      // // ??? TEMPORARY DIAGNOSTIC CODE
+      // // Compute the cost of the order, as delivered (in PO currency), not including shipping+handling
+      // const costDelivered = movement.lots.reduce((accum, lot) => accum + lot.unit_cost * lot.quantity, 0.0);
+
+      // Compute the ratio of the original PO shipping+handling costs to the total cost of lots
+      // (Note that the total shipping_handling is copied to each lot in originalPO)
+      const SHRatio = vm.originalPO[0].shipping_handling / costPO;
+
+      // // ??? TEMPORARY DIAGNOSITIC CODE
```

Comment:
Can this be removed?

Reply:
I will remove these commented-out diagnostic items once the PR has been reviewed for correct operation.